src/lib/handleAuth/handleAuth.ts

```diff
@@ -13,6 +16,9 @@ export async function handleAuth({
 	let url: URL | null = null;
 	switch (params.kindeAuth) {
 		case 'login':
+			if (options.post_login_redirect_url && typeof options.post_login_redirect_url == 'string') {
+				sessionStorage.setSessionItem(KEY_POST_LOGIN_URL, options.post_login_redirect_url);
+			}
 			url = await kindeAuthClient.login(request as unknown as SessionManager, options);
 			break;
 		case 'register':
```

Comment:
Oh let's remember the dynamic redirect URL here as well, before registration. This is also done by the [NextJS SDK](https://github.com/kinde-oss/kinde-auth-nextjs/blob/cd517020b57c257c4fa44c07ba9668f0028bbd2d/src/handlers/register.js#L20-L23). It's perhaps even more important for new customers to have a smooth sign-up flow that drops them right back where they were in their user journey.